pkg/operation/botanist/health_check.go

```diff
@@ -49,7 +50,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func mustGardenRoleLabelSelector(gardenRoles ...string) labels.Selector {
+func mustGardenRoleLabelSelector(gardenRoleKey string, gardenRoles ...string) labels.Selector {
```

Comment:
```suggestion
func mustLabelSelector(gardenRoleKey string, gardenRoles ...string) labels.Selector {
```

`gardenRoleKey` parameter is not used.